packages/flutter/lib/src/material/checkbox_list_tile.dart

```diff
@@ -57,6 +57,165 @@ import 'theme_data.dart';
 /// ```
 /// {@end-tool}
 ///
+/// ## Semantics in CheckboxListTile
+///
+/// Since CheckboxListTile is wrapped with a [MergeSemantics] widget, it wants
+/// to merge its children semantics nodes into one node in the semantics tree.
+/// Therefore, CheckboxListTile would not work with [RichText], which forces
+/// itself to be a standalone semantics node.
+///
+/// If you're interested in learning more about this interaction, see the
```

Comment:
I don't see anything about Semantics in https://docs.flutter.io/flutter/painting/TextSpan/recognizer.html

Maybe leave this paragraph out?

Reply:
This line cause it to become independent semantics node https://github.com/flutter/flutter/blob/4e1bfca8470ff088ecb68aaa72086f290b316fd1/packages/flutter/lib/src/rendering/paragraph.dart#L504

Reply:
Yeah, I knew it was in `RenderParagraph`, but I just didn't know how to correlate the ideas well, since `RenderParagraph` is utilized by `RichText` under the hood and I didn't want the reader to have to dig too deep to get the idea. I'm leaning on just leaving it out and giving a high level explanation.